src/ui/components/SecondaryToolbox/react-devtools/__tests__/utils.ts

```diff
@@ -0,0 +1,160 @@
+import { createElement } from "react";
+import type { Store } from "react-devtools-inline";
+import { createRoot } from "react-dom/client";
+
+import { deconstructOperationsArray, reconstructOperationsArray } from "../rdtProcessing";
+
+export function assertContainsOperationType(operationsArrays: number[][], type: number) {
+  for (let index = 0; index < operationsArrays.length; index++) {
+    const operationsArray = operationsArrays[index];
+    const operationsObject = deconstructOperationsArray(operationsArray);
+    if (operationsObject.treeOperations.find(operation => operation.type === type)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+export function assertContainsOperationTypes(operationsArrays: number[][], types: number[]) {
+  for (let index = 0; index < types.length; index++) {
+    const type = types[index];
+    if (!assertContainsOperationType(operationsArrays, type)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+export function verifySerialization(operationsArrays: number[][]) {
+  operationsArrays.forEach(operationsArray => {
+    expect(operationsArray.length > 0).toBeTruthy();
+
+    const deconstructed = deconstructOperationsArray(operationsArray);
+    const reconstructed = reconstructOperationsArray(deconstructed);
+
+    expect(reconstructed).toEqual(operationsArray);
+  });
+}
```

Comment:
I think these helper functions could be used in the #9576 approach even if we don't go with this PR.